Added ARO full cycle pipeline

trepel opened the pull request:
## Overview

Full cycle (provision + install + test + cleanup) pipeline for ARO (OpenShift on Azure)

### What has been done

Added full cycle ARO pipeline.

Added support for using existing ARO cluster via `provision-cluster` param -> the cluster and related resources (`az create ...` commands) are only created if this is set to 'true'. The simple pipeline we have for ARO provisioning has this param hard coded to 'true' -> tbh I am thinking about removing this pipeline because this new full cycle pipeline makes it a bit redundant.

The `provision-aro` Task also updated to
* use latest version if no version is specified via `create-cmd-flags` param
* create secret with cluster credentials but only if cluster is provisioned as part of pipeline run

## Verification Steps

Trigger the pipeline. However, simpler might be to just see the pipeline runs in trepel ns.

averevki commented on main/aro-provision-install-test-pipeline.yaml:
```diff
@@ -0,0 +1,325 @@
+        name: delete-aro
+      workspaces:
+        - name: shared-workspace
+    - name: cluster-secret-cleanup
```
This task doesn't use workspace with saved kubeconfig and removes secret from already removed cluster? How does that work here?

trepel replied:
This pipeline runs on some cluster (let's call it Hub Cluster). It provisions ARO cluster and creates a secret with credentials to this (newly provisioned) ARO cluster in Hub Cluster. Once ARO cluster is removed this credentials secret in Hub Cluster is of no use anymore and it also needs to be removed as part of cleanup.

TaskRuns have access to cluster they are running on (Hub Cluster) automatically so no need to setup any kubeconfig.

averevki replied:
Makes sense, thanks for the explanation

averevki reviewed (commented):
I'd also remove ARO pipelines it replaces here, gj :+1:

trepel commented on main/aro-provision-install-test-pipeline.yaml:
```diff
@@ -0,0 +1,325 @@
+      description: Prefix of the launch name saved in report portal (nightly, username, manual, etc.). In case of release candidate testing use kuadrant-v<version>, rhcl-v<version>, or authorino-v<version>
+      name: launch-name
+      type: string
+    - default: 'Pipeline'
```
Good point, will update. Thanks!

trepel commented:
> I'd also remove ARO pipelines it replaces here, gj 👍

I think that all the `provision-<some-infra>` pipelines are a bit redundant once we have "full cycle" pipelines for each infra. I would remove all these at once in a separate PR (this one is big enough without it imo).

trepel commented:
@averevki Thanks for the review! I added some fixes and improvement so the 2nd commit to make it easier for you to see the changes and re-review. I can squash the commits if you think that it would be better. What has been done in 2nd commit:
- fixed defaults as you suggested - in all full cycle pipelines, not just this one
- added operator restart Task into the pipeline (it was part of "hardening PR" you reviewed recently and it makes sense to have it for this full cycle ARO pipeline as well
- check-image-existence Task is triggered immediately now - in all full cycle pipelines
- fixed helm-flags and helm-tools flags since these parameters have been changed recently - in all full cycle pipelines

Please re-review if you can spare some time, thanks!

averevki commented on main/aro-provision-install-test-pipeline.yaml:
```diff
@@ -0,0 +1,325 @@
+        - name: shared-workspace
+    - name: check-image-existence
+      when:
+        - input: "$(params.index-image)"
+          operator: notin
+          values: ['']
```
This actually works fine, sorry

averevki reviewed (approved):
LGTM
